### PR TITLE
Restructure driftsort write-up and refactor introduction

### PR DESCRIPTION
**Pull request by orlp**

### CODE REVIEW

File: writeup/driftsort_introduction/text.md [hunk 1]
```diff
@@ -40,92 +42,146 @@ The primary goal was to develop a replacement for the current Rust standard libr
...
+- **Ancestor pivot tracking**: By keeping track of earlier used pivots in Quicksort we can detect and process common elements, allowing for O(N * log(K)) total comparisons where K is the number of distinct elements. This is also great for Zipfian distributions.
+- **Partition**: Branchless stable partition. Left-to-right scan that performs a `cmov` style pointer select to fill the auxiliary memory buffer from left-to-right or right-to-left depending on the comparison outcome.
+- **Mergesort**: Mergesort fallback used when a recursion depth limit is reached in the Quicksort. Re-uses the existing merging code in driftsort by calling itself with a flag that forces small runs to be made with insertion sort.
+
+### Components from glidesort which are not incorporated
+
+Some components/techniques from glidesort were intentionally not incorporated because they were considered too niche or have too much codesize overhead for a generic standard library sort:
+
+- **Merge splitting.** This allows arbitrarily small auxiliary buffers which was not necessary for `slice::sort` and enables interleaved merge loops which were not implemented for codesize reasons.
```
Comment (Voultapher) [inline writeup/driftsort_introduction/text.md:83]: I think we should settle one of either binary-size or codesize, and stay consistent with it.

File: writeup/driftsort_introduction/text.md [hunk 2]
```diff
@@ -40,92 +42,146 @@ The primary goal was to develop a replacement for the current Rust standard libr
...
+- **Quad-merges.** Delaying merges for quad-merges with interleaved loops and avoiding `memcpy`s was not implemented as the interleaved loops were too large and the `memcpy` relatively cheap.
+- **Bidirectional partitioning.** This was rather complex and a large amount of code to avoid relatively cheap `memcpy`s in the stable partition, and the interleaved loops were larger.
```
Comment (Voultapher) [inline writeup/driftsort_introduction/text.md:85]: IIRC it was also arch dependent whether that even improved run-time, right?

**Review by Voultapher — APPROVED** [review]
Looks good, only minor things.

File: writeup/driftsort_introduction/text.md (hunk 2 above)
Comment (orlp) [reply writeup/driftsort_introduction/text.md:85]: Yes and no, the loop interleaving was hit/miss depending on the CPU, but a non-interleaved loop version could still fully avoid the `memcpy` partition we ended up settling on.

**Pull request merged**